ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/BlockSizeBlockValidator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright contributors to Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import static org.hyperledger.besu.evm.gascalculator.OsakaGasCalculator.MAX_RLP_BLOCK_SIZE;
+
+import org.hyperledger.besu.ethereum.BlockProcessingResult;
+import org.hyperledger.besu.ethereum.MainnetBlockValidator;
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.core.Block;
+
+/** Validates blocks ensuring they do not exceed the maximum allowed size. */
+public class BlockSizeBlockValidator extends MainnetBlockValidator {
+
+  /**
+   * Constructs a new MainnetBlockValidator with the given BlockHeaderValidator, BlockBodyValidator,
+   * BlockProcessor, and BadBlockManager.
+   *
+   * @param blockHeaderValidator the BlockHeaderValidator used to validate block headers
+   * @param blockBodyValidator the BlockBodyValidator used to validate block bodies
+   * @param blockProcessor the BlockProcessor used to process blocks
+   */
+  public BlockSizeBlockValidator(
+      final BlockHeaderValidator blockHeaderValidator,
+      final BlockBodyValidator blockBodyValidator,
+      final BlockProcessor blockProcessor) {
+    super(blockHeaderValidator, blockBodyValidator, blockProcessor);
+  }
+
```

Comment:
Since the limit is in the gasCalculator "for now", this class can be removed and the check can be moved to MainnetTransactionValidator.

Reply:
Rather keep it as is as in the future in this PR or another the limit being in the GasCalculator will be removed and at that point I'd want this in it's own validator.